src/main/java/org/benetech/servicenet/filters/RequestFilter.java

```diff
@@ -0,0 +1,120 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.Ordered;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+import org.springframework.web.util.WebUtils;
+
+@Component
+public class RequestFilter extends OncePerRequestFilter implements Ordered {
+
+    @Autowired
+    private RequestLoggerService requestLoggerService;
+
+    @Autowired
+    private RequestLoggerMapper requestLoggerMapper;
+
+    @Override
+    public int getOrder() {
+        return Ordered.LOWEST_PRECEDENCE - 8;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+        throws ServletException, IOException {
+        ContentCachingRequestWrapper wrappedRequest = new ContentCachingRequestWrapper(request);
+        ContentCachingResponseWrapper wrappedResponse = new ContentCachingResponseWrapper(response);
+
+        filterChain.doFilter(wrappedRequest, wrappedResponse);
+
+        Map<String, Object> trace = getTrace(wrappedRequest);
+        Map<String, Object> responseTrace = getResponseTrace(wrappedResponse);
+
+        getBody(wrappedRequest, trace);
+        getResponseBody(wrappedResponse, responseTrace);
+
+        RequestLogger requestLogger = new RequestLogger();
+        requestLogger.setRemoteAddr(Objects.toString(trace.get("remoteAddr"), ""));
+        requestLogger.setRequestUri(Objects.toString(trace.get("requestUri"), ""));
+        requestLogger.setRequestMethod(Objects.toString(trace.get("requestMethod"), ""));
+        requestLogger.setRequestParameters(Objects.toString(trace.get("requestParams"), ""));
+        requestLogger.setRequestBody(Objects.toString(trace.get("body"), ""));
+        requestLogger.setResponseStatus(Objects.toString(responseTrace.get("status"), ""));
+        requestLogger.setResponseBody(Objects.toString(responseTrace.get("body"), ""));
+
+        requestLoggerService.save(requestLoggerMapper.toDto(requestLogger));
+
+        wrappedResponse.copyBodyToResponse();
+    }
+
+    private void getBody(ContentCachingRequestWrapper request, Map<String, Object> trace) {
+        ContentCachingRequestWrapper wrapper = WebUtils.getNativeRequest(request, ContentCachingRequestWrapper.class);
+        if (wrapper != null) {
+            byte[] buf = wrapper.getContentAsByteArray();
+            if (buf.length > 0) {
+                String payload;
+                try {
+                    payload = new String(buf, 0, buf.length, wrapper.getCharacterEncoding());
+                }
+                catch (UnsupportedEncodingException ex) {
+                    payload = "[unknown]";
+                }
+
+                trace.put("body", payload.replaceAll("\"password\":\".+?\"", "\"password\":\"fakePassword\""));
```

Comment:
I wonder if there is more sensitive data; for example we should also filter out the newly changed passwords (I believe this field is called newPassword?)

Reply:
Is this replace fast enough for big requests? maybe we could only do replacing for some endpoints that we know have this sensitive data

Reply:
Replace now will work for authentication and for change-password endpoints.